### PR TITLE
Feature/grafana agent replacement

### PR DESCRIPTION
**Pull request by MadsHT**
**Description of your changes:**
We have decided that Grafana-agents was to imature to use for the time being.
This made us return to the Kube-Prometheus-stack.
This has been updated to the latest version and configured so that it can look up the serviceMonitors with the `instance: primary` label, so no changes are needed to those (for now).

Promtail has been added as a seperate chart, this was done in case we ever want to return the Grafana agents, then it's easy to disable Promtail without having to reconfigure loki.

I also set a retention on both Loki and Cortex, to keep storage usage down while keeping some historical data.

When this PR is completed I will open [Yggdrasil#76](https://github.com/distributed-technologies/yggdrasil/pull/76) PR for review.

Checklist:
* [x] I have bumped the chart version
* [x] I have documented my changes if this is needed
* [x] I have described my changes in the "description of your changes" field
* [x] I have tested this change on a running cluster and the Argocd dashboard shows healthy and synced
* [ ] I have created the necessary tests in the chart, if they are possible/necessary
* [x] I have squashed commits if necessary

### CODE REVIEW

**Comment by klausenbusk** [comment]
Can we downgrade prometheus-operator to < v0.52.0? AFAIU the CRD grow too big with that release.

**Comment by MadsHT** [comment]
Sound fine to me, I'll downgrade to [chart version 19.3.0](https://github.com/prometheus-community/helm-charts/blob/kube-prometheus-stack-19.3.0/charts/kube-prometheus-stack/Chart.yaml) since > [20.0.0](https://github.com/prometheus-community/helm-charts/blob/kube-prometheus-stack-20.0.0/charts/kube-prometheus-stack/Chart.yaml) is app version > 0.52.0

File: charts/prometheus/chart/Chart.yaml [hunk 1]
```diff
@@ -2,9 +2,12 @@ apiVersion: v2
 name: prometheus
 description: A Helm chart for Prometheus
 type: application
-version: 0.1.3
+version: 0.1.4
 
+# In version 33.1.0 there are some problems with a too big CRD from the operator, see https://github.com/prometheus-operator/prometheus-operator/issues/4439#issue-1073133029
+# For now I will not update to the latest version.
+# @KlausenBusk Found that around appVersion 0.52.0 the CRD grew, so we should not upgrade beyond 0.52.0 until ArgoCD 2.3.0 is released.
```
[the remark this answers is not in the document]
Comment (MadsHT) [reply charts/prometheus/chart/Chart.yaml:9]: Makes sense (y)

File: charts/loki/chart/values.yaml [hunk 2]
```diff
@@ -1,2 +1,5 @@
 # https://github.com/grafana/helm-charts/blob/main/charts/loki-distributed/values.yaml
-# loki-distributed:
+loki-distributed:
+  config:
+    limits_config:
+      retention_period: 336h
```
Comment (klausenbusk) [inline charts/loki/chart/values.yaml:5]: Does this work? Retention isn't enabled in the compactor (`retention_enabled` AFAIU).

https://grafana.com/docs/loki/latest/operations/storage/retention/

**Comment by klausenbusk** [comment]
I would prefer the last commit to be split into two and add some context to the `Add promtail proxy chart` commit (why is it added?).

File: charts/loki/chart/values.yaml (hunk 2 above)
Comment (MadsHT) [reply charts/loki/chart/values.yaml:5]: No, you're right, well spotted

**Comment by klausenbusk** [comment]
Nitpicks, but the rest looks good:
- `promtail: Add Promtail proxy chart` -> `Add Promtail proxy chart` (we kinda agreed to drop the prefix to the initial chart commit)
- `prometheus: Updated Prometheus proxy chart` -> `prometheus: Update Prometheus proxy chart` (imperative)

**Comment by MadsHT** [comment]
No problem, I forgot the init chart thing, and the second was an oversight

**Review by klausenbusk — DISMISSED** [review]
LGTM 🎉

File: charts/promtail/chart/values.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+promtail:
+  initContainer:
+    # -- Specifies whether the init container for setting inotify max user instances is to be enabled
+    enabled: true
+    # -- The inotify max user instances to configure
+    fsInotifyMaxUserInstances: 512
```
Comment (skovbakke) [inline charts/promtail/chart/values.yaml:6]: Why do we set `fsInotifyMaxUserInstances` to 512?

**Review by skovbakke — COMMENTED** [review]
A few remarks.

Could you look at the last commit message again? There is a couple of spelling mistakes that confused me, eg `futher` --> `further`, and `untroduced` --> `introduced`.

File: charts/promtail/chart/values.yaml (hunk 3 above)
Comment (MadsHT) [reply charts/promtail/chart/values.yaml:13]: I have added a comment about it

**Review by skovbakke — APPROVED** [review]
LGTM

**Review by klausenbusk — APPROVED** [review]
LGTM 🎉

**Pull request merged**